src/Core/Resolvers/SqlPaginationUtil.cs

```diff
@@ -234,45 +260,56 @@ public static JsonDocument CreatePaginationConnectionFromJsonDocument(JsonDocume
+        public static IEnumerable<PaginationColumn> ParseAfterFromJsonString(
+            string afterJsonString,
+            PaginationMetadata paginationMetadata,
+            ISqlMetadataProvider sqlMetadataProvider,
+            string entityName,
+            RuntimeConfigProvider runtimeConfigProvider
+            )
         {
-            IEnumerable<PaginationColumn>? after;
+            List<PaginationColumn>? after = new();
+            IEnumerable<NextLinkColumn>? requestAfter;
             try
             {
                 afterJsonString = Base64Decode(afterJsonString);
-                after = JsonSerializer.Deserialize<IEnumerable<PaginationColumn>>(afterJsonString);
+                requestAfter = JsonSerializer.Deserialize<IEnumerable<NextLinkColumn>>(afterJsonString);
 
-                if (after is null)
+                if (requestAfter is null)
                 {
                     throw new ArgumentException("Failed to parse the pagination information from the provided token");
                 }
 
                 Dictionary<string, PaginationColumn> afterDict = new();
-                foreach (PaginationColumn column in after)
+                foreach (NextLinkColumn column in requestAfter)
                 {
                     // REST calls this function with a non null sqlMetadataProvider
                     // which will get the exposed name for safe messaging in the response.
                     // Since we are looking for pagination columns from the $after query
                     // param, we expect this column to exist as the $after query param
                     // was formed from a previous response with a nextLink. If the nextLink
                     // has been modified and backingColumn is null we throw exception.
-                    string backingColumnName = GetBackingColumnName(entityName, column.ColumnName, sqlMetadataProvider);
+                    string backingColumnName = GetBackingColumnName(entityName, column.FieldName, sqlMetadataProvider);
                     if (backingColumnName is null)
                     {
-                        throw new DataApiBuilderException(message: $"Cursor for Pagination Predicates is not well formed, {column.ColumnName} is not valid.",
-                                                       statusCode: HttpStatusCode.BadRequest,
-                                                       subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                        throw new DataApiBuilderException(
+                            message: $"Cursor for Pagination Predicates is not well formed, {column.FieldName} is not valid.",
+                            statusCode: HttpStatusCode.BadRequest,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
                     }
 
+                    PaginationColumn pageColumn = new(
+                        tableName: "",
```

Comment:
When parsing the $after string, is it okay to not populate the actual table and schema name for further processing?

Reply:
it turns out that we don't actually generate the table name in the query from this information. we get the table name from the lookup we do from the request information itself.

Reply:
If that is the case, then why do we need `NextLinkColumn.EntityName` ? since the entity name will be present in the request URL anyway?

Reply:
that's there for clarification, it was what davide suggested, and it makes sense to include since the information is safe to expose and could be helpful if youre tracking down an issue related to pagination.

Reply:
is the tableName not required even when `PaginationColumn` is used for the GraphQL case?

Reply:
That's right, it isn't needed for either REST or GraphQL.